pie/storage/database.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+from typing import Dict, Optional, Union
+
+from sqlalchemy import BigInteger, Column, String
+
+from pie.database import database, session
+
+
+class Storage(database.base):
+    __tablename__ = "pie_storage_data"
+
+    module = Column(String, primary_key=True)
+    guild_id = Column(BigInteger, primary_key=True)
+    key = Column(String, primary_key=True)
+    value = Column(String)
+    type = Column(String)
+
+    @staticmethod
+    def set(
+        module: str,
+        guild_id: int,
+        key: str,
+        value,
+        allow_overwrite: bool = True,
+    ) -> Optional[Storage]:
+        data = (
+            session.query(Storage)
+            .filter_by(module=module)
+            .filter_by(guild_id=guild_id)
+            .filter_by(key=key)
+            .one_or_none()
+        )
+
+        if data and not allow_overwrite:
+            return None
+
+        if not data:
+            data = Storage(module=module, key=key, guild_id=guild_id)
+
+        data.value = value
+        data.type = type(value)
```

Comment:
Does it work? Or does it need `type(value).__name__`?

Reply:
We have to extract the name since we are passing this value to inspect.